### PR TITLE
feat: add config for max fee estimations

### LINKED ISSUE #2039: Add fee estimation caps to the config file
At the moment fees are being capped. 

By adding a toggle to the config file we'd have an easy way to add or remove those caps without having to go through a wallet release.

### PR DESCRIPTION
**Pull request by beguene**
> Try out this version of the Hiro Wallet - download [extension builds](https://github.com/hirosystems/stacks-wallet-web/actions/runs/1777134026).

closes #2039

Added a config in wallet-config.json to 
- enable/disable fee estimations max values
- set custom values for those max values

cc/ @kyranjamie @fbwoolf

### CODE REVIEW

File: config/wallet-config.json [hunk 1]
```diff
@@ -2,5 +2,13 @@
   "activeFiatProviders": {
     "transak": { "name": "transak", "enabled": true },
     "okcoin": { "name": "okcoin", "enabled": false }
+  },
+  "feeEstimations": {
+    "maxValuesEnabled": true,
+    "maxValues": {
```
Comment (kyranjamie) [inline config/wallet-config.json]: Maybe this could be an array to model the fee estimation API response?

**Review by kyranjamie — APPROVED** [review]
👍🏼LGTM

File: src/common/transactions/transaction-utils.ts [hunk 2]
```diff
@@ -49,47 +47,6 @@ function txHasTime(tx: Tx) {
   );
 }
 
-const maxValueForLowFeeEstimation = 500000;
```
Comment (fbwoolf) [inline src/common/transactions/transaction-utils.ts]: Nice idea to move all this into its own file. 👍

**Review by fbwoolf — APPROVED** [review]
Nice implementation @beguene. Looks good. 👍

**Comment by fbwoolf** [comment]
Trying to get these integration tests to pass so I can merge this. They pass for me locally so maybe we need to address the flakiness of these tests at some point?

**Comment by kyranjamie** [comment]
Please don't merge, until my PR goes in 🙏🏼

**Comment by fbwoolf** [comment]
> Please don't merge, until my PR goes in 🙏🏼

Yep, np, I thought we were trying to get these in first. I can just leave this for Beguene to finish up.

File: src/app/query/hiro-config/hiro-config.query.ts [hunk 3]
```diff
@@ -54,3 +61,17 @@ export function useHasFiatProviders() {
     Object.keys(activeProviders).reduce((acc, key) => activeProviders[key].enabled || acc, false)
   );
 }
+
+export function useConfigFeeEstimationsEnabled() {
+  const config = useRemoteHiroConfig();
+  if (typeof config?.feeEstimations === 'undefined') return;
```
Comment (kyranjamie) [inline src/app/query/hiro-config/hiro-config.query.ts]: In `app/utils` we have `isNumber` & `isString` methods. Maybe you can add one for undefined, too.

Comment (beguene) [reply src/app/query/hiro-config/hiro-config.query.ts]: Sure !

**Comment by andresgalante** [comment]
@beguene Can we merge this one?

**Comment by beguene** [comment]
@andresgalante I have to fix the conflicts and test again. Do we still need this given the new changes regarding fee estimations in the blockchain ?

**Comment by andresgalante** [comment]
Yes, I want a safety net in case the new estimations fail we can quickly set up caps without having to release. Can we merge it please?

**Pull request merged**